### PR TITLE
New: Added new Drop Shadow group control [OBXT-567]

### PR DESCRIPTION
**Pull request by louiswol94**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

New: Added new Drop Shadow group control.

### CODE REVIEW

File: includes/controls/groups/drop-shadow.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	 * @since 3.7.0
+	 * @access protected
+	 *
+	 * @return array Default drop shadow control options.
+	 */
+	protected function get_default_options() {
+		return [
+			'popover' => [
+				'starter_title' => _x( 'Drop Shadow', 'Drop Shadow Control', 'elementor' ),
+				'starter_name' => 'drop_shadow_type',
```
Comment (udidol) [inline includes/controls/groups/drop-shadow.php:85]: Don't think the `_type` suffix is necessary

**Review by udidol — CHANGES_REQUESTED** [review]

File: includes/controls/groups/drop-shadow.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	 * @static
+	 *
+	 * @var array drop shadow control fields.
+	 */
+	protected static $fields;
+
+	/**
+	 * Get drop shadow control type.
+	 *
+	 * Retrieve the control type, in this case `box-shadow`.
```
[the remark this answers is not in the document]
Comment (louiswol94) [reply includes/controls/groups/drop-shadow.php]: @udidol Typo fixed.
Done.

File: includes/controls/groups/drop-shadow.php (hunk 1 above)
Comment (louiswol94) [reply includes/controls/groups/drop-shadow.php:85]: @udidol `_type` suffix is necessary. You will see `box shadow` and `text shadow` also has these suffixes.

https://github.com/elementor/elementor/blob/master/includes/controls/groups/text-shadow.php#L86
https://github.com/elementor/elementor/blob/master/includes/controls/groups/box-shadow.php#L97

Without this suffix the control stops working.

File: assets/dev/js/editor/editor-base.js [hunk 3]
```diff
@@ -178,6 +178,7 @@ export default class EditorBase extends Marionette.Application {
 			Url: require( 'elementor-controls/url' ),
 			Wp_widget: require( 'elementor-controls/wp_widget' ),
 			Wysiwyg: require( 'elementor-controls/wysiwyg' ),
+			Drop_shadow: require( 'elementor-controls/box-shadow' ),
```
[the remark this answers is not in the document]
Comment (louiswol94) [reply assets/dev/js/editor/editor-base.js:181]: @udidol Yes it works 100%. You will see `Text Shadow` also just uses the `Box Shadow` JS as it does the same work. So it was not needed to create a new JS file that has the same functionality.

https://github.com/elementor/elementor/blob/master/assets/dev/js/editor/editor-base.js#L177

Maybe we can rename `box-shadow.js` to something more generic like `shadows.js`?

File: includes/controls/groups/drop-shadow.php (hunk 1 above)
Comment (udidol) [reply includes/controls/groups/drop-shadow.php:85]: I see, ok cool good to know

File: assets/dev/js/editor/editor-base.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (udidol) [reply assets/dev/js/editor/editor-base.js:181]: I see. No need to change it just leave it as is. Don't want to cause any trouble for potential 3rd parties.

**Review by udidol — APPROVED** [review]

File: assets/dev/js/editor/editor-base.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (matipojo) [reply assets/dev/js/editor/editor-base.js:181]: Hi, Please explain it in the code.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/editor-base.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (louiswol94) [reply assets/dev/js/editor/editor-base.js:186]: Hi @matipojo, I added a comment in the code to explain this.

[the remark this answers is not in the document]
Comment (matipojo) [reply assets/dev/js/editor/editor-base.js:186]: Thanks!

File: includes/controls/drop-shadow.php [hunk 4]
```diff
@@ -0,0 +1,118 @@
+<?php
```
Comment (matipojo) [inline includes/controls/drop-shadow.php:1]: Can it extend the box-shadow control?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/controls/drop-shadow.php (hunk 4 above)
Comment (louiswol94) [reply includes/controls/drop-shadow.php:1]: Hi @matipojo 
Apparently this task has been put on hold for now :-)

**Pull request closed without merging**